code/dijkstra-algorithm/ViewModel/src/main/java/ru/unn/agile/dijkstraalgorithm/viewmodel/ViewModel.java

```diff
@@ -0,0 +1,159 @@
+package ru.unn.agile.dijkstraalgorithm.viewmodel;
+
+import javafx.beans.binding.BooleanBinding;
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.beans.property.SimpleStringProperty;
+import javafx.beans.property.StringProperty;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import ru.unn.agile.dijkstraalgorithm.model.DijkstraGraph;
+
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class ViewModel {
+
+    private DijkstraGraph graph;
+
+    private static final Pattern VERTEX_INPUT_ALLOWED_SYMBOLS = Pattern.compile("^[a-zA-Z]+$");
+    private static final Pattern WEIGHT_INPUT_ALLOWED_SYMBOLS = Pattern.compile("^[1-9][0-9]*$");
+    private final SimpleBooleanProperty addingNewEdgeDisabled = new SimpleBooleanProperty();
+
+
+    private final StringProperty vertex1 = new SimpleStringProperty();
+    private final StringProperty vertex2 = new SimpleStringProperty();
+    private final StringProperty weight = new SimpleStringProperty();
+
+    private StringProperty vertexFrom = new SimpleStringProperty();
+    private StringProperty vertexTo = new SimpleStringProperty();
+    private StringProperty resultPath = new SimpleStringProperty();
+
+
+    private final ObservableList<EdgeViewModel> edgeList = FXCollections.observableArrayList();
+    private final ObservableList<String> vertexList = FXCollections.observableArrayList();
+
+
+    public ViewModel() {
+        clearFormInput();
+
+        BooleanBinding canCalculateBoolBinding = new BooleanBinding() {
+            {
+                super.bind(vertex1, vertex2, weight);
+            }
+            @Override
+            protected boolean computeValue() {
+                return (isVertex1InputCorrect()
+                        && isVertex2InputCorrect()
+                        && isWeightInputCorrect());
+            }
+        };
+        addingNewEdgeDisabled.bind(canCalculateBoolBinding.not());
+    }
+
+    public void addEdge() {
+        EdgeViewModel newEdge = new EdgeViewModel(vertex1.get(), vertex2.get(), weight.get());
+        edgeList.add(newEdge);
+
+        clearFormInput();
+    }
+
+    private void clearFormInput() {
+        vertex1.set("");
+        vertex2.set("");
+        weight.set("");
+    }
+
+    public boolean isVertex1InputCorrect() {
+        String exprText1 = vertex1.get();
+        String exprText2 = vertex2.get();
+        return (VERTEX_INPUT_ALLOWED_SYMBOLS.matcher(exprText1).matches()
+                && !(exprText1.equals(exprText2)));
+    }
+
+    public boolean isVertex2InputCorrect() {
+        String exprText1 = vertex1.get();
+        String exprText2 = vertex2.get();
+        return (VERTEX_INPUT_ALLOWED_SYMBOLS.matcher(exprText2).matches()
+                && !(exprText1.equals(exprText2)));
+    }
```

Comment:
Maybe put the second check in a separate method? 
Something like areVertexInputsAreEquals(..)

Reply:
Done.